dio/lib/src/transformer.dart

```diff
@@ -89,7 +89,7 @@ class DefaultTransformer extends Transformer {
   @override
   Future transformResponse(
       RequestOptions options, ResponseBody response) async {
-    if (options.responseType == ResponseType.stream) {
+    if (options.responseType == ResponseType.stream || response.statusCode==204) {
```

Comment:
May be better to check if the response is an empty string.

Reply:
my response data is null but on line 164(responseBody = utf8.decode(responseBytes, allowMalformed: true);) executed change my response data  null to empty string

Reply:
> change my response data null to empty string

Yeah that's another annoying issue.